### PR TITLE
feat(molecule/modal): added molecule modal

### PR DESCRIPTION
**Pull request by ferransimon**
Created a modal component following the guide

### CODE REVIEW

File: components/molecule/modal/src/index.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+   */
+  open: PropTypes.bool,
+  /**
+   * OnClose function handler
+   */
+  onClose: PropTypes.func
+}
+
+MoleculeModal.defaultProps = {
+  centerVertically: false,
```
Comment (jcger) [inline components/molecule/modal/src/index.js]: this should be true

File: components/molecule/modal/src/index.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+   */
+  header: PropTypes.element,
+  /**
+   * customitzable close icon
+   */
+  iconClose: PropTypes.element.isRequired,
+  /**
+   * prop to mark if the modal is currently open or not
+   */
+  open: PropTypes.bool,
```
Comment (jcger) [inline components/molecule/modal/src/index.js]: what about calling it `isOpen` in order to make it more readable?

File: components/molecule/modal/src/index.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this._onKeyDown)
+  }
+
+  _onKeyDown = event => {
+    if (this.state.open === false || this.props.closeOnEscKeyDown === false)
+      return
+    if (SUPPORTED_KEYS.includes(event.key)) {
```
Comment (jcger) [inline components/molecule/modal/src/index.js]: FYI: `includes` doesn't work in IE11 https://caniuse.com/#search=includes, but we have a polyfill for it https://github.com/SUI-Components/sui/blob/master/packages/sui-polyfills/src/index.js#L5, so no worries 👍

File: components/molecule/modal/src/index.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    })
+
+    const dialogClassName = cx(suitClass({element: 'dialog'}), {
+      [suitClass({element: 'dialog--full'})]: this.props.fitWindow
+    })
+
+    return (
+      <div
+        className={wrapperClassName}
+        ref={node => {
```
Comment (jcger) [inline components/molecule/modal/src/index.js]: please use the React.createRef() API as callbacks ref was used by older react versions

File: components/molecule/modal/src/index.scss [hunk 5]
```diff
@@ -0,0 +1,109 @@
+@import '~@schibstedspain/sui-theme/lib/settings-compat-v7/index';
```
Comment (jcger) [inline components/molecule/modal/src/index.scss]: we need to remove this dependency

File: components/molecule/modal/src/index.scss [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    box-shadow: $bxsh-modal-basic;
+    display: flex;
+    flex-direction: column;
+    margin: $m-modal-basic;
+    max-height: 100%;
+    text-align: left;
+    vertical-align: middle;
+    white-space: normal;
+
+    @include media-breakpoint-down(s) {
```
Comment (jcger) [inline components/molecule/modal/src/index.scss]: we need to remove this media query, more info https://github.schibsted.io/scmspain/frontend-all-wiki/wiki/Breakpoints

File: demo/molecule/modal/playground [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+          isOpen={this.state.open}
+          closeOnOutsideClick
+          closeOnEscKeyDown
+          iconClose={<IconClose />}
+          fitWindow
+          header={<strong>My new brand modal</strong>}
+          onClose={this.closeModal}
+        >
+          <div>
+              <p>
```
Comment (jcger) [inline demo/molecule/modal/playground]: you could create a component with this paragraph and reuse it

File: components/molecule/modal/src/index.scss [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    padding: 0;
+    position: relative;
+  }
+
+  &-close {
+    background-color: transparent;
+    border: 0;
+    cursor: pointer;
+    display: flex;
+    padding: $pt-molecule-modal-header $pr-molecule-modal-header $pb-molecule-modal-header $pl-molecule-modal-header;
```
Comment (jcger) [inline components/molecule/modal/src/index.scss]: isn't there a conflict with the padding set by the header?

File: components/molecule/modal/src/index.scss [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+  &.is-MoleculeModal-open {
+    display: flex;
+    justify-content: center;
+    overflow-x: hidden;
+    overflow-y: auto;
+  }
+
+  &-dialog {
+    background-color: $bg-moleule-modal-dialog;
```
Comment (jcger) [inline components/molecule/modal/src/index.scss]: I'm missing the border-radius set in the visuals

File: components/molecule/modal/src/index.js [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+class MoleculeModal extends Component {
+  _contentRef = React.createRef()
+  _wrapperRef = React.createRef()
+
+  componentDidMount() {
+    document.addEventListener('keydown', this._onKeyDown)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this._onKeyDown)
```
Comment (jcger) [inline components/molecule/modal/src/index.js:43]: we need to add `this._toggleWindowScroll(false)`

File: package.json [hunk 11]
```diff
@@ -27,9 +27,9 @@
   "author": "",
   "license": "ISC",
   "devDependencies": {
-    "@s-ui/deploy": "2",
+    "@s-ui/deploy": "^2.9.0",
```
Comment (jcger) [inline package.json]: please let this as it was

File: components/molecule/modal/src/index.scss (hunk 8 above)
Comment (nguasch) [reply components/molecule/modal/src/index.scss]: I'm very confused by `paddings`

File: components/molecule/modal/src/index.js (hunk 10 above)
Comment (ferransimon) [reply components/molecule/modal/src/index.js:43]: done!

File: demo/molecule/modal/playground [hunk 12]
```diff
@@ -0,0 +1,320 @@
+const IconClose = () => <span>X</span>
```
[the remark this answers is not in the document]
Comment (ferransimon) [reply demo/molecule/modal/playground]: done

File: components/molecule/modal/src/index.js [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+        className={suitClass({element: 'header'})}
+        onTouchMove={e => e.preventDefault()}
+      >
+        {header}
+        {this._renderCloseIcon()}
+      </div>
+    )
+  }
+
+  _renderNoHeader = () => {
```
[the remark this answers is not in the document]
Comment (ferransimon) [reply components/molecule/modal/src/index.js]: done

File: demo/molecule/modal/playground (hunk 7 above)
Comment (ferransimon) [reply demo/molecule/modal/playground]: done!

File: package.json [hunk 14]
```diff
@@ -29,7 +29,7 @@
   "devDependencies": {
     "@s-ui/deploy": "2",
     "@s-ui/precommit": "2",
-    "@s-ui/studio": "5",
+    "@s-ui/studio": "^5.18.0",
```
Comment (jcger) [inline package.json]: please undo this changes

File: components/molecule/modal/src/index.scss [hunk 15]
```diff
@@ -1,31 +1,80 @@
 @import '~@schibstedspain/sui-theme/lib/index';
 @import './settings';
 
+@mixin modal-position {
```
Comment (devictoribero) [inline components/molecule/modal/src/index.scss]: Since this modal-position is something static and is not changing, I would suggest to change it from `@mixin` to `%placeholder` due to it reduces the final `css bundle` or on the other hand, `@mixin` duplicates the `css`

File: components/molecule/modal/src/index.scss [hunk 16]
```diff
@@ -1,31 +1,80 @@
...
+  &-out::before {
+    animation: modal-overlay-out .25s both;
+  }
 
   &.is-MoleculeModal-open {
     display: flex;
     justify-content: center;
     overflow-x: hidden;
-    overflow-y: auto;
+    overflow-y: auto;
```
Comment (devictoribero) [inline components/molecule/modal/src/index.scss]: extra space at the end of the line

**Review by jcger — DISMISSED** [review]

File: components/molecule/modal/src/index.js [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+    document.addEventListener('keydown', this._onKeyDown)
+    this._wrapperRef.current.addEventListener(
+      'webkitAnimationEnd',
+      this.props.onAnimationEnd
+    )
+  }
+  componentWillUnmount() {
+    toggleWindowScroll(false)
+    document.removeEventListener('keydown', this._onKeyDown)
+    this._wrapperRef.current.addEventListener(
```
Comment (midudev) [inline components/molecule/modal/src/index.js]: You could just use https://reactjs.org/docs/events.html#animation-events.

`onAnimationEnd` on the wrapper div.

Comment (ferransimon) [reply components/molecule/modal/src/index.js]: done!

**Review by jcger — DISMISSED** [review]

**Review by jcger — APPROVED** [review]

**Review by nguasch — APPROVED** [review]
